src/main/scala/io/iohk/ethereum/consensus/ConsensusImpl.scala

```diff
@@ -63,26 +61,19 @@ class ConsensusImpl(
     *   - [[DuplicateBlock]]     - block already exists either in the main chain or in the queue
     *   - [[BlockImportFailed]]  - block failed to execute (when importing to top or reorganising the chain)
     */
-  override def evaluateBranchBlock(
-      block: Block
-  )(implicit blockExecutionScheduler: Scheduler, blockchainConfig: BlockchainConfig): Task[BlockImportResult] =
+  override def evaluateBranch(
+      branch: Seq[Block]
+  )(implicit blockExecutionScheduler: Scheduler, blockchainConfig: BlockchainConfig): Task[BlockImportResult] = {
+    val block = branch.head
     blockchainReader.getBestBlock() match {
```

Comment:
The `blockchainReader.getBestBlock()` was also moved to `ConsensusAdapter`, so I think you can remove it from here and simplify the code

Reply:
I did not want to make the assumption that the caller will provide the best block, so I think we should keep it this way even if it is indeed a bit wasteful.